File: 4008-RestoreFinishingOrder/4008-RestoreFinishingOrder.java
// Last updated: 3/30/2026, 2:10:54 PM
class Solution {
    public int[] recoverOrder(int[] order, int[] friends) {
        int N = order.length, F = friends.length;
        int[] result = new int[F];
        int ctr = 0;
        for(int i = 0 ; i < N ; i++){
            for(int j = 0 ; j < F ; j++){
                if(order[i] == friends[j]){
                    result[ctr] = order[i];
                    ctr++;
                }
            }
        }
        return result;
    }
}

File: 4008-RestoreFinishingOrder/4008-RestoreFinishingOrder.cpp
// Last updated: 3/30/2026, 2:10:58 PM
class Solution {
public:
    vector<int> recoverOrder(vector<int>& order, vector<int>& friends) {
        vector<int> result;
        unordered_map<int,int> hashmap;
        int N = order.size(), F = friends.size();
        for(int index = 0 ; index < N ; index++){
            if(find(friends.begin(),friends.end(),order[index]) != friends.end()) result.push_back(order[index]);
        }
        return result;
    }
};